### PR TITLE
[Doc] Add cache select doc

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/zh/data_source/data_cache_warmup.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+在数据湖分析过程中，有一些场景对查询有一定的性能要求，比如 BI 报表，性能测试 POC 等。可以提前将远端数据载入 Data Cache，避免查询时还需要从远端拉取数据，从而提供快速、稳定的查询性能。
+
+## CACHE SELECT 语法
+
+StarRocks 提供了 `CACHE SELECT` 语法来实现 Data Cache 预热。使用 `CACHE SELECT` 之前，需要确保已经开启 [Data Cache](./data_cache.md) 特性。
+
+`CACHE SELECT` 的语法如下：
+
+```sql
+CACHE SELECT column_name [, ...] FROM table_name [WHERE boolean_expression] [PROPERTIES("verbose"="true")]
```
[the remark this answers is not in the document]
Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache_warmup.md:18]: 就和正常查表一样，你可以制定 Full QualifiedName，也可以只指定 table 名称。

如果只指定 table 名称，那么会从当前会话的 db 下面搜索。

File: docs/zh/data_source/data_cache_warmup.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
+---
+displayed_sidebar: "Chinese"
+---
+
+# Data Cache 预热
+
+本文介绍如何通过 Data Cache 预热 (Warmup）来提前将远端数据载入 Data Cache。
+
+在数据湖分析过程中，有一些场景对查询有一定的性能要求，比如 BI 报表，性能测试 POC 等。可以提前将远端数据载入 Data Cache，避免查询时还需要从远端拉取数据，从而提供快速、稳定的查询性能。
```
[the remark this answers is not in the document]
Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache_warmup.md:9]: Data Cache 是一个被动填充cache的过程，相当于你在查询的时候，顺带把数据写入 cache。
但是 cache select 是一个主动填充 cache 的过程。cache select 更像是基于 data cache 做的扩展。

File: docs/zh/data_source/data_cache_warmup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (evelynzhaojie) [reply docs/zh/data_source/data_cache_warmup.md:18]: data cache 文档里之前写的是 sr 原生表不生效

File: docs/zh/data_source/data_cache_warmup.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+| SUCCESS | 15.2MB              | 754.7MB          | 88.7ms               | 29.04%            |
++---------+---------------------+------------------+----------------------+-------------------+
+1 row in set (36.56 sec)
+```
+
+- `STATUS`：预热任务的执行结果。
+- `ALREADY_CACHED_SIZE`：Data Cache 中已缓存的数据大小（目前这块统计存在一定误差，后续会改进）。
+- `WRITE_CACHE_SIZE`：写入 Data Cache 的大小。
+- `AVG_WRITE_CACHE_TIME`：每一个文件写入 Data Cache 的平均耗时。
+- `TOTAL_CACHE_USAGE`：本次预热执行完成后 Data Cache 的空间使用率，可以根据这个指标评估 Data Cache 的空间是否充足。
```
Comment (evelynzhaojie) [inline docs/zh/data_source/data_cache_warmup.md:51]: 可以根据这个指标评估 Data Cache 的空间是否充足。--- 用户怎样评估，空间使用率高于多少，就不充足了？

Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache_warmup.md:51]: 这个没有标准答案。比如你 warm up 1gb 的表，那你剩余空间有 1g 就行了。

File: docs/zh/data_source/data_cache_warmup.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+   1 row in set (0.03 sec)
+   ```
+
+## 使用限制和说明
+
+* 需要开启 Data Cache 特性，且拥有对目标 catalog/database/table 的 SELECT 权限。
+* `CACHE SELECT` 支持存算分离和存算一体架构的外表查询，支持预热远端的 TEXT, ORC, Parquet 文件。
+* `CACHE SELECT` 只支持对单表进行预热，不支持 `ORDER BY`，`LIMIT`，`GROUP BY` 等算子。
+* 目前 CACHE SELECT 的实现是采用 `INSERT INTO BLACKHOLE()` 的方案，即按照正常的查询流程对表进行预热。所以 `CACHE SELECT` 的性能开销和普通查询的开销是差不多的。这一块后续会做出改进，提升 `CACHE SELECT` 的性能。
+* `CACHE SELECT` 预热的数据不会保证一定不被淘汰，Data Cache 底层仍然按照 LRU 规则进行淘汰。用户可以自行通过 `SHOW BACKENDS\G` 查看 Data Cache 的剩余容量，以此判断是否会触发 LRU 淘汰。
```
[the remark this answers is not in the document]
Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache_warmup.md:176]: SHOW BACKENDS\G 里面会显示 DataCache 的空间使用情况。如果空间满了，那就会有 LRU 淘汰发生。

File: docs/zh/data_source/data_cache_warmup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Smith-Cruise) [reply docs/zh/data_source/data_cache_warmup.md]: 是的，只有外表生效。

**Comment by evelynzhaojie** [comment]
updated in this new PR https://github.com/StarRocks/starrocks/pull/44673/files because error occurs when I pull this pr to local and edit

**Pull request closed without merging**